docs/sql.md

```diff
@@ -0,0 +1,148 @@
+---
+sql:
+  gaia: ./lib/gaia-sample.parquet
+---
+
+# SQL
+
+Observable Framework includes built-in support for client-side SQL powered by [DuckDB](./lib/duckdb). You can use SQL to query data from [CSV](./lib/csv), [TSV](./lib/csv), [JSON](./javascript/files#json), [Apache Arrow](./lib/arrow), and [Apache Parquet](./lib/arrow#apache-parquet) files. These files can either be static or generated by [data loaders](./loaders).
+
+## SQL front matter
+
+To use SQL, first register the desired tables in the page’s [front matter](./markdown#front-matter) using the **sql** option. This option takes an object, where each key is a table name and each value is the to the corresponding data file. For example, to register a table named `gaia` with data from a Parquet file:
+
+```yaml
+---
+sql:
+  gaia: ./lib/gaia-sample.parquet
+---
+```
+
+## SQL code blocks
+
+To run SQL queries, create a SQL fenced code block (<code>```sql</code>). For example, to query the first 10 rows from the `gaia` table:
+
+````md
+```sql
+SELECT * FROM gaia ORDER BY phot_g_mean_mag LIMIT 10
+```
+````
+
+This produces a table:
+
+```sql
+SELECT * FROM gaia ORDER BY phot_g_mean_mag LIMIT 10
+```
+
+You can interpolate dynamic values into SQL queries using inline expressions `${…}`. For example, to show the stars around a given brightness:
+
+```js echo
+const mag = view(Inputs.range([6, 20], {label: "Magnitude"}));
+```
+
+```sql echo
+SELECT * FROM gaia WHERE phot_g_mean_mag BETWEEN ${mag - 0.1} AND ${mag + 0.1};
+```
+
+To refer to the results of a query in JavaScript, use the `id` directive. For example, to refer to the results of the previous query as `top10`:
+
+````md
+```sql id=top10
+SELECT * FROM gaia ORDER BY phot_g_mean_mag LIMIT 10
+```
+````
+
+```sql id=top10
+SELECT * FROM gaia ORDER BY phot_g_mean_mag LIMIT 10
+```
+
+This returns an array of 10 rows:
+
+```js echo
+top10
+```
+
+The `id` directive supports destructuring, so you can refer to individual rows and columns. For example, to pull out the top row:
+
+````md
+```sql id=[top]
+SELECT * FROM gaia ORDER BY phot_g_mean_mag LIMIT 1
+```
+````
+
+```sql id=[top]
+SELECT * FROM gaia ORDER BY phot_g_mean_mag LIMIT 1
+```
+
+```js echo
+top
+```
+
+To pull out the minimum value of the `phot_g_mean_mag` column:
+
+````md
+```sql id=[{min}]
+SELECT MIN(phot_g_mean_mag) AS min FROM gaia
+```
```

Comment:
I would have liked this to be extensible to define multiple values, but it doesn't seem to work:

````md
```sql id=[{min, max}]
SELECT MIN(phot_g_mean_mag) AS min, MAX(phot_g_mean_mag) AS max FROM gaia
```
````

defines only `min`.

(Not a blocker.)

Reply:
Hmm, that’s surprising. I would expect that to work and I’ll investigate.

Reply:
In fact it works, I was editing the wrong code block! 😭

Reply:
We could update that example in the documentation to include min and max?

Reply:
Ah, so the attribute gotcha applies, which means you either need quotes

````md
```sql id="[{min, max}]"
SELECT MIN(phot_g_mean_mag) AS min, MAX(phot_g_mean_mag) AS max FROM gaia
```
````

or to remove spaces

````md
```sql id=[{min,max}]
SELECT MIN(phot_g_mean_mag) AS min, MAX(phot_g_mean_mag) AS max FROM gaia
```
````